source_control/gh_keys.py

```diff
@@ -0,0 +1,177 @@
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import requests
+from ansible.module_utils.basic import *
+
+DOCUMENTATION = '''
+---
+module: gh_keys
+version_added: n/a
+short_description: Manages github ssh keys.
+description:
+    - The module manages the ssh key for a specific user through Github API v3.
+options:
+  action:
+    description: This tells the gh_keys module what you want it to do.
+    required: true
+    choices: 
+      list_keys:
+        description: lists keys for authenticated or non-authenticated user. If the passwd field is informed, the module returns all keys and all data about its (In this case, the module will send the authentication data in request). Otherwise, the module returns only some data about the ssh keys of non-authenticated user.
+        required_fields: [user | user & passwd]
+      get_key:
+        description: gets a specific key by github ssh-key-id
+        required_fields: [user & passwd & key_id]
+      add_key:
+        description: adds a new public ssh key
+        required_fields: [user & passwd & title & key]
+      remove_key:
+        description: removes a specific key by github ssh-key-id
+        required_fields: [user & passwd & key_id]
+  user:
+    description: Github username.
+    required: true
+  passwd:
+    description: Github password. If 2FA is enabled for your account, you should generate a new personal access token. 
+    required: for actions: list_keys (your), get_key, add_key, remove_key
+  title:
+    description: Title of the new ssh key
+    required: for add_key action
+  key:
+    description: The path of file that contains the public key
+    required: for add_key action
+  key_id:
+    description: The key id provided by Github
+    required: for get_key and remove_key actions
+author: Leonardo Comelli
+'''
+
+EXAMPLES = '''
+# Example from Ansible Playbooks
+
+# Lists limit informations about all keys of non-authenticated user
+- gh_keys: action=list_keys user=leocomelli
+
+# Lists all informations about all keys of authenticated user
+- gh_keys: action=list_keys user=leocomelli passwd=secret
+
+# Gets a single public key (authetication required)
+- gh_keys: action=get_key user=leocomelli passwd=secret key_id=8767854
+
+# Adds a new public key (authentication required)
+- gh_keys: action=add_key user=leocomelli passwd=secret title=my_new_key key=/home/leocomelli/.ssh/id_rsa.pub
+
+# Removes an existing ssh key
+- gh_keys: action=remove_key user=leocomelli passwd=secret key_id=8767854
+'''
+
+GH_API_URL = "https://api.github.com/%s"
+
+class GHKeys(object):
+
+  def __init__(self, module):
+    self.action = module.params['action']
+    self.user   = module.params['user']
+    self.passwd = module.params['passwd']
+    self.title  = module.params['title']
+    self.key    = module.params['key']
+    self.key_id = module.params['key_id']
+  
+  def work(self):
```

Comment:
To me, this function looks a little too "clever" and should perhaps be made a bit more simplistic and explicit.